address-controller/src/main/java/io/enmasse/controller/common/KubernetesHelper.java

```diff
@@ -344,6 +352,106 @@ public boolean isDestinationClusterReady(String clusterId) {
         return Optional.ofNullable(client.secrets().inNamespace(namespace).withName(secretName).get());
     }
 
+    private JsonObject doRawHttpRequest(String path, String method, JsonObject body, boolean errorOk) {
+        OkHttpClient httpClient = client.adapt(OkHttpClient.class);
+
+        HttpUrl url = HttpUrl.get(client.getOpenshiftUrl()).resolve(path);
+        Request request = new Request.Builder()
+                .url(url)
+                .addHeader("Content-Type", "application/json")
+                .addHeader("Authorization", "Bearer " + controllerToken)
+                .method(method, body != null ? RequestBody.create(MediaType.parse("application/json"), body.encode()) : null)
+                .build();
+        try (Response response = httpClient.newCall(request).execute()) {
+            if (response.isSuccessful()) {
+                return new JsonObject(response.body().string());
+            } else {
+                if (errorOk) {
+                    response.close();
+                    return null;
+                } else {
+                    String errorMessage = String.format("Error performing %s on %s: %d, %s", method, path, response.code(), response.body());
+                    log.warn(errorMessage);
+                    throw new RuntimeException(errorMessage);
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+
+    @Override
+    public TokenReview performTokenReview(String token) {
+        if (client.isAdaptable(OkHttpClient.class)) {
+            JsonObject body = new JsonObject();
+
+            body.put("kind", "TokenReview");
+            body.put("apiVersion", "authentication.k8s.io/v1beta1");
+
+            JsonObject spec = new JsonObject();
+            spec.put("token", token);
+            body.put("spec", spec);
+
+            JsonObject responseBody= doRawHttpRequest("/apis/authentication.k8s.io/v1beta1/tokenreviews", "POST", body, false);
+            JsonObject status = responseBody.getJsonObject("status");
+            boolean authenticated = false;
+            String userName = null;
+            if (status != null) {
+                authenticated = status.getBoolean("authenticated");
+                JsonObject user = status.getJsonObject("user");
+                if (user != null) {
+                    userName = user.getString("username");
+                }
+            }
+            return new TokenReview(userName, authenticated);
+        } else {
+            return new TokenReview(null, false);
+        }
+    }
+
+    @Override
+    public SubjectAccessReview performSubjectAccessReview(String user, String verb) {
+        if (client.isAdaptable(OkHttpClient.class)) {
+            JsonObject body = new JsonObject();
+
+            body.put("kind", "SubjectAccessReview");
```

Comment:
Because you're now checking access to namespaced resources, you need to use LocalSubjectAccessReview here.